### PR TITLE
Migrate InventorySync incl. ITs

### PR DESCRIPTION
**Pull request by salander85**
JIRA: https://commercetools.atlassian.net/browse/DEVX-136

### CODE REVIEW

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/InventoryServiceImplIT.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            inventoryEntry -> {
+              assertThat(inventoryEntry.getQuantityOnStock()).isEqualTo(QUANTITY_ON_STOCK_2);
+              assertThat(inventoryEntry.getRestockableInDays()).isEqualTo(RESTOCKABLE_IN_DAYS_2);
+              assertThat(inventoryEntry.getExpectedDelivery()).isEqualTo(EXPECTED_DELIVERY_2);
+            });
+
+    // assert CTP state
+    final Optional<InventoryEntry> updatedInventoryEntry =
+        getInventoryEntryBySkuAndSupplyChannel(CTP_TARGET_CLIENT, SKU_2, null, null);
+    assertThat(updatedInventoryEntry).isEqualTo(result.get());
```
Comment (lojzatran) [inline src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/InventoryServiceImplIT.java:137]: This cannot be equal. One is type Optional and one is type InventoryEntry.

File: src/main/java/com/commercetools/sync/sdk2/inventories/InventorySync.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+      "Failed to update inventory entry of SKU '%s' and " + "supply channel id '%s'.";
+  private static final String FAILED_TO_PROCESS =
+      "Failed to process the InventoryEntryDraft with SKU:'%s'. " + "Reason: %s";
+
+  private final InventoryService inventoryService;
+  private final InventoryReferenceResolver referenceResolver;
+  private final InventoryBatchValidator batchValidator;
+
+  /**
+   * Takes a {@link com.commercetools.sync.inventories.InventorySyncOptions} instance to instantiate
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/inventories/InventorySync.java:56]:
```suggestion
   * Takes a {@link InventorySyncOptions} instance to instantiate
```

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/externalsource/inventories/InventorySyncIT.java [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+                CTP_SOURCE_CLIENT, REFERENCE_ID_TO_KEY_CACHE, inventoryEntries)
+            .join();
+
+    // Prepare sync options and perform sync of draft to target project.
+    final InventorySyncOptions inventorySyncOptions =
+        InventorySyncOptionsBuilder.of(CTP_TARGET_CLIENT).ensureChannels(true).build();
+    final InventorySync inventorySync = new InventorySync(inventorySyncOptions);
+    final InventorySyncStatistics inventorySyncStatistics =
+        inventorySync.sync(newInventories).toCompletableFuture().join();
+    assertThat(inventorySyncStatistics).hasValues(3, 1, 1, 0);
```
[the remark this answers is not in the document]
Comment (salander85) [reply src/integration-test/java/com/commercetools/sync/integration/sdk2/externalsource/inventories/InventorySyncIT.java:344]: [97f9f3e](https://github.com/commercetools/commercetools-sync-java/pull/1034/commits/97f9f3eded421e7094b88edc5d178e7cd5241b0b)

File: src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/InventoryServiceImplIT.java (hunk 1 above)
Comment (salander85) [reply src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/InventoryServiceImplIT.java]: [a399ee4](https://github.com/commercetools/commercetools-sync-java/pull/1034/commits/a399ee48648787ac19e6609b5ab925a8c39bdde4)

**Comment by salander85** [comment]
> Please check the code coverage drop.
The main changes without tests are coming from other migrated sync modules, e.g. statesync. I think this will be fixed when merging all together.

**Pull request merged**